### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by joreeves** [sisoe24/NukeServerSocket#23]
**Describe the bug**
Strange behavior with multiple function definitions. Trying to use a function defined in the same script inside of another function returns ```An exception occurred running Nuke Internal engine: `name 'first_function' is not defined` ```

**To Reproduce**
version: Nuke 13.2v2
Having more than 1 function defined in the script and call them from another function.
```
import nuke

def first_function():
    print('first_function')

def second_function():
    print('second_function')

def test():
    first_function()
    second_function()

first_function() # This will run
second_function() # This will run
test() # This will fail
```

**Expected behavior**
I would expect this to behave the same as the script editor in nuke.

**Comment by sisoe24** [comment]
Hey! Thanks for reaching out.

This issue has been discussed before #21, and the problem is the execution engine. The Nuke Internal uses the `exec` method, which is not the most trustworthy. I did not investigate further since I planned to remove the feature altogether. 

But I gave it a second chance now, and passing the `globals()` function as an argument to `exec` "seems" to do the job, albeit not sure if it has other side effects. 

If you want to give it a go, this is the location: https://github.com/sisoe24/NukeServerSocket/blob/819e807bc6860881206af2e67b8fbcd36b6ea331/src/controllers/nuke_controllers.py#L67

I am still unsure if keeping around `exec` is a good idea, but I prefer the Nuke intended API for this job rather than the other method (Script Editor Engine), which feels like a hack.

Working code sample

```python
import io
import sys
import contextlib


@contextlib.contextmanager
def stdoutIO(stdout=None):
    old = sys.stdout
    if stdout is None:
        stdout = io.StringIO()

    sys.stdout = stdout
    yield stdout
    sys.stdout = old


def _exec(data):
    with stdoutIO() as s:
        exec(data, globals())  # without the globals(), the exec fails
    return s.getvalue()


code = '''
def first_function():
    print('first_function')
first_function()

def second_function():
    print('second_function')
second_function()

def test():
    first_function()
    second_function()

test()
'''

print("result :", _exec(code))
```

**Comment by sisoe24** [comment]
I added an empty `dict` to `exec('str', {})`, which seems to do the job better than adding the `globals()` function. I am still figuring out if this is the best solution, so I want to avoid pushing a hotfix.

I will include the fix in the next release (#24) unless happens in between.

I am closing this issue for now but feel free to open it again.
